### ISSUE TITLE
Context needs to be better passed through to several parts of Zarf so that interrupts can be implemented properly

### ISSUE THREAD

**Issue by AustinAbro321** [defenseunicorns/zarf#2594]
Many parts of Zarf do not properly handle context properly. Since #2567 was merged in ctrl-c may not interrupt as a user would expect in several situations. 
At least
- pulling doesn't get interrupted 
- During waits on deploy rather than exiting the Zarf binary interrupting exits just the current wait
- Interrupts work on-create actions, but give a message `failed after 0 retries` rather than `package creation canceled`

**Comment by AustinAbro321** [comment]
duplicate of  #2647
